ssh-proxy/orb.yml

```diff
@@ -0,0 +1,72 @@
+version: 2.1
+description: An orb using sshuttle to establish a proxy to a remote network for performing deployments
+# Some of this Orb have been inspired by https://circleci.com/developer/orbs/orb/symbol/ssh-tunnel (MIT Licence)
+
+commands:
+  proxy_up:
+    description: "Establish a routed sshuttle proxy over an SSH tunnel to the remote network"
+    parameters:
+      hostname:
+        description: "The hostname of the SSH proxy server"
+        type: string
+      port:
+        description: "The port of the SSH proxy server"
+        type: integer
+      username:
+        description: "The username to connect to the SSH proxy server"
+        type: string
+      private_key_fingerprint:
+        description: "The private key for connecting to the SSH proxy server. The key must be added via CircleCI settings first: https://circleci.com/docs/2.0/add-ssh-key/"
+        type: string
+      routed_network_cidrs:
+        description: "A string of CIDR blocks requests for which should be proxied, separated by a space. For individual IP addresses, use x.x.x.x/32 to represent them"
+        type: string
+      connection_test_host:
+        description: "The destination IP or hostname in the private network which we want to reach via the proxy, for testing the connection"
+        type: string
+      connection_test_port:
+        description: "The port for the host in the private network which we want to reach via the proxy, for testing the connection"
+        type: integer
+
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - << parameters.private_key_fingerprint >>
+      - run:
+          name: Install additional packages required
+          shell: /bin/bash
+          command: |
+            sudo apt update && sudo apt install -y sshuttle
```

Comment:
This is a Ubuntu package in the official repository: https://packages.ubuntu.com/xenial/sshuttle

Not installing this in a Dockerfile because this proxy needs to run on a VM in CircleCI instead of a container anyway.